src/Provisioning.ts

```diff
@@ -87,6 +87,20 @@ export class Provisioner {
         return (currentCount >= this.main.config.provisioning?.limits?.room_count);
     }
 
+    private async determineSlackIdForRequest(matrixUserId, teamId) {
+        const matrixUser = await this.main.datastore.getMatrixUser(matrixUserId);
+        if (matrixUser === null) {
```

Comment:
I prefer `(!matrixUser)`.
btw. getMatrixUser's signatures don't match but TypeScript doesn't complain. Some implementations return undefined, others null.

![Screenshot_2020-05-07_15-12-18](https://user-images.githubusercontent.com/10872136/81298595-43208700-9075-11ea-91b0-1dc655314773.png)

Reply:
Ah, because MatrixUser is any, so essentially types are disabled here.